### PR TITLE
py-sfepy: update to v2021.3

### PR DESCRIPTION
**Pull request by rc**

### CODE REVIEW

**Comment by alalazo** [comment]
@adamjstewart Wondering why spackbot didn't run on this PR :thinking:

**Comment by rc** [comment]
> @adamjstewart Wondering why spackbot didn't run on this PR thinking

I disabled github actions on my fork, to prevent running all the checks for every commit prior to the PR. Might that be related?

**Comment by rc** [comment]
I have just fixed the tarball url...

File: var/spack/repos/builtin/packages/py-sfepy/package.py [hunk 1]
```diff
@@ -14,11 +14,14 @@ class PySfepy(PythonPackage):
     """
 
     homepage = "https://sfepy.org"
-    url      = "https://github.com/sfepy/sfepy/archive/release_2017.3.tar.gz"
+    url      = "https://github.com/sfepy/sfepy/archive/release_2021.3.tar.gz"
 
-    version('2017.3', sha256='d13642b7abed63b83b7eaef4dfce6e84a5afc5798bc7ffa1c413e3e44b5e5996')
+    version('2021.3', sha256='822407595e0af5aeaca6ca50e88e46658b5021e177244cbb763dd260e94eda81')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-sfepy/package.py:20]: Spack's policy is not to remove old versions without first deprecating them to give users time to migrate to the new release, see https://spack.readthedocs.io/en/latest/packaging_guide.html#deprecating-old-versions

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-sfepy/package.py (hunk 1 above)
Comment (rc) [reply var/spack/repos/builtin/packages/py-sfepy/package.py:20]: Thanks for pointing that out - I also tried to make the new dependencies conditional based on the version. Conditional variants are not available yet, right?

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-sfepy/package.py:20]: Correct, although you can add conflicts that prevent a variant from being used for older versions. I think @becker33 was working on conditional variants?

File: var/spack/repos/builtin/packages/py-sfepy/package.py [hunk 2]
```diff
@@ -14,11 +14,16 @@ class PySfepy(PythonPackage):
     """
 
     homepage = "https://sfepy.org"
-    url      = "https://github.com/sfepy/sfepy/archive/release_2017.3.tar.gz"
+    url      = "https://github.com/sfepy/sfepy/archive/release_2021.3.tar.gz"
+    git      = "https://github.com/sfepy/sfepy.git"
 
-    version('2017.3', sha256='d13642b7abed63b83b7eaef4dfce6e84a5afc5798bc7ffa1c413e3e44b5e5996')
+    version('2017.3', sha256='d13642b7abed63b83b7eaef4dfce6e84a5afc5798bc7ffa1c413e3e44b5e5996', deprecated=True)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-sfepy/package.py]: Can you sort from newest to oldest?

Comment (rc) [reply var/spack/repos/builtin/packages/py-sfepy/package.py]: Sure!

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**